sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -238,16 +238,16 @@ async def sub_test_non_releasing_messages():
                 receiver = sb_client.get_queue_receiver(servicebus_queue.name)
                 sender = sb_client.get_queue_sender(servicebus_queue.name)
 
-                def _hack_disable_receive_context_message_received(self, message):
-                    # pylint: disable=protected-access
-                    self._handler._was_message_received = True
-                    self._handler._received_messages.put(message)
+                # def _hack_disable_receive_context_message_received(self, message):
+                #     # pylint: disable=protected-access
+                #     self._handler._was_message_received = True
+                #     self._handler._received_messages.put(message)
```

Comment:
Not sure what this was originally testing and why it had to be removed?

Reply:
This was originally testing overloading the receive_mixins() enhanced_receive_message() function that releases functions if the receive_context is not set. This function just no longer released messages so you were testing that trying to complete expired messages would error, and then trying to re-receive them after with an increased delivery count . 

I commented this out because it did not seem like lines 249-250 were actually resetting the value of on_message_received and so messages were still being released.

Reply:
hmm so the test was always broken?